### ISSUE TITLE
scrolling doesn't work in some apps

### ISSUE THREAD

**Issue by amosbird** [Genymobile/scrcpy#820]
I assume that `scrolling != finger swiping` which leads to various issues. Is it possible to make mouse scrolling fake swiping?

**Comment by rom1v** [comment]
Is this question really different from #433?

> Is there any reliable way to generate swipe left/up/down/right events using scrcpy?

Scrolling sends scroll events to Android. I think this is the correct behavior. If apps don't handle scroll events, then they don't scroll.

**Comment by amosbird** [comment]
> Is this question really different from #433?

Yes, #443 asked for key bindings to fake swiping. In this issue, I'm seeking some way of making scrolling sends swipe events instead of scroll events. I guess that means multiple events are fired consecutively thus faking a mouse down, mouse move, and move up won't work

**Comment by instasck** [comment]
adb input will have not real touch so some apps might avoid it

**Comment by Zero3K** [comment]
Any news regarding this issue?

**Comment by rom1v** [comment]
Does e2d5f0e7fc98af97f3785ce492b7e8f87b109df1 fix the issue (test the current `dev` branch).

**Comment by Binly42** [comment]
With v1.16, scroll still doesn't work in some apps like `qidian`(a novel reader), possibly for the reason that those apps usually protectively ban/customize some interacting (e.g. long-press to select&copy text ...)

**Comment by rom1v** [comment]
Or they just use a custom component where they didn't implement scrolling (by reacting to scroll events).
